src/core/src/main/java/org/apache/jmeter/util/keystore/JmeterKeyStore.java

```diff
@@ -312,22 +280,18 @@ private int getIndexAndIncrement(int length) {
      * TODO Currently, keyType and issuers are both ignored.
      *
      * @param keyType the key algorithm type name (RSA, DSA, etc.)
-     * @param issuers  the CA certificates we are narrowing our selection on.
-     *
-     * @return the array of aliases; may be empty
+     * @param issuers the CA certificates we are narrowing our selection on.
+     * @return the array of aliases; may be null
      */
     public String[] getClientAliases(String keyType, Principal[] issuers) {
         int count = getAliasCount();
-        String[] aliases = new String[count];
-        for(int i = 0; i < aliases.length; i++) {
-            aliases[i] = this.names[i];
-        }
-        if(aliases.length>0) {
-            return aliases;
-        } else {
+        if (count == 0) {
             // API expects null not empty array, see http://docs.oracle.com/javase/7/docs/api/javax/net/ssl/X509KeyManager.html
-            return null; // NOSONAR
+            return null;
         }
-    }
+        String[] aliases = new String[count];
+        System.arraycopy(this.names, 0, aliases, 0, aliases.length);
```

Comment:
Is this `Arrays.copyOf(...)`?

Reply:
Good point that would be an even better change.

Reply:
Ah, `Arrays.copyOf` only works for primitives i.e. not Strings. We could do:
`return = Arrays.stream(this.names).toArray(String[]::new);` or will have to leave it as it is.

Reply:
Does this work? https://docs.oracle.com/javase/7/docs/api/java/util/Arrays.html#copyOf(T[],%20int)

Reply:
At least in jshell (java 11) the last one seems to work when given the length of the original array. Another alternative would be to use `aliases.clone()` (*duck*)

Reply:
Used Arrays.copyOf with the length, but also noticed `getCertificateChain` wasn't behaving as per the docs, so have updated it.